src/iotivity/iotivity_api.js

```diff
@@ -0,0 +1,841 @@
+
+  //if (g_iotivity_device)
+  //  g_iotivity_device.configure(this.settings);
+}
+
+// partial dictionary is ok
+// equivalent to ‘onboard+configure’ in Core spec
+// maps to IoTivity Configure (C++ API), configure (Java API), OCInit (C API)
+OicDevice.prototype.configure = function(settings) {
+  DBG('OicDevice.configure=' + JSON.stringify(settings));
+  if (settings)
+    _addConstProperty(this, 'settings', new OicDeviceSettings(settings));
+
+  var msg = {
+    'cmd': 'configure',
+    'settings': {
+      'url': this.settings.url,
+      'role': this.settings.role,
+      'connectionMode': this.settings.connectionMode,
+      'info': this.settings.info
+    }
+  };
+
+  return createPromise(msg);
+};
+
+// return to factory configuration and reboot
+OicDevice.prototype.factoryReset = function() {
+  var msg = {
+    'cmd': 'factoryReset'
+  };
+  return createPromise(msg);
+};
+
+// keep configuration and reboot
+OicDevice.prototype.reboot = function() {
+  var msg = {
+    'cmd': 'configure',
+    'settings': settings
+  };
+  return createPromise(msg);
+};
+
+iotivity.OicDevice = OicDevice;
+
+///////////////////////////////////////////////////////////////////////////////
+// OicDeviceSettings
+///////////////////////////////////////////////////////////////////////////////
+
+function OicDeviceSettings(obj) {
+  DBG('new OicDeviceSettings: obj=' + JSON.stringify(obj));
+  if (obj) {
+    _addConstProperty(this, 'url', obj.url);
+    _addConstProperty(this, 'info', new OicDeviceInfo(obj.info));
+    _addConstProperty(this, 'role', obj.role);
+    _addConstProperty(this, 'connectionMode', obj.connectionMode);
+  }
+  else {
+    _addConstProperty(this, 'url', '0.0.0.0:0');
+    _addConstProperty(this, 'info', new OicDeviceInfo(null));
+    _addConstProperty(this, 'role', 'intermediate');
+    _addConstProperty(this, 'connectionMode', 'acked');
+  }
+}
+
+iotivity.OicDeviceSettings = OicDeviceSettings;
+
+///////////////////////////////////////////////////////////////////////////////
+// OicDeviceInfo
+///////////////////////////////////////////////////////////////////////////////
+
+function OicDeviceInfo(obj) {
+  DBG('new OicDeviceInfo: obj=' + JSON.stringify(obj));
+  if (obj) {
+    _addConstProperty(this, 'uuid', obj.uuid);
+    _addConstProperty(this, 'name', obj.name);
+    _addConstProperty(this, 'dataModels', obj.dataModels);
+    _addConstProperty(this, 'coreSpecVersion', obj.coreSpecVersion);
+    _addConstProperty(this, 'model', obj.model);
+    _addConstProperty(this, 'manufacturerName', obj.manufacturerName);
+    _addConstProperty(this, 'manufacturerUrl', obj.manufacturerUrl);
+    _addConstProperty(this, 'manufacturerDate', obj.manufacturerDate);
+    _addConstProperty(this, 'platformVersion', obj.platformVersion);
+    _addConstProperty(this, 'firmwareVersion', obj.firmwareVersion);
+    _addConstProperty(this, 'supportUrl', obj.supportUrl);
+  }
+  else {
+    _addConstProperty(this, 'uuid', 'default');
```

Comment:
This behaviour has not been specified. Should it be covered in the Web API spec?